### PR TITLE
Add jest test coverage

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- Added steps to CI Github action to download, process and upload coverage reports for jest tests (stable and beta)
- Updated jest and rush configurations to get json json-summary coverage reports
- Updated test:coverage actions for rush. In order to use it, run `rushx test:coverage` for a package. For packages inside `packages` folder a new `temp/jest/coverage/` folder will be created and HTML report will be generated. Please note that reported will be rewritten if you generate it for an another package.
- Fixed 1 of the warnings in tests
- If an html report is needed for the whole `azure/communication-react`:
1. Download coverage reports from PR artifacts
1. Update  all`/home/runner/work/communication-ui-library` in `detailed-full/coverage-report.json` to a folder where `communication-react` project is located locally.
1. Run in terminal`npx nyc report --temp-dir '<path-to-jest-coverage-report-{flavor}-folder>/detailed-full/' --report-dir <directory-path-where-html-report-will-be-saved> --reporter lcov --exclude-after-remap false`
1. After the last step, an html report will be created and index.html file should be used for coverage preview.

Please also refer to 'Jest Tests Coverage Spike' document for more details on steps.

CI time:
- `compare_jest_tests_coverage` and `update_jest_coverage_report` adds ~10-15 seconds to the CI pipeline (for both flavors)
- `build_packages` time is increased by ~1 min for each flavor

UPD: After adding a script for calculations, `compare_jest_tests_coverage` time is increased to ~1 min for each flavor because of dependencies install

# Why

Added functionality for jests coverage reports generation

# How Tested

CI and local testing

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -674,6 +686,129 @@ jobs:
...
+        run: curl -o temp/jest/base/coverage-report.json "https://gist.github.com/communication-ui-bot/${{ matrix.gist_id }}/raw/communication-react-jest-report-${{ matrix.flavor }}.json" --create-dirs -L
+      - name: Download coverage report from the build_packages job
+        uses: actions/download-artifact@v3
+        with:
+          name: jest-coverage-report-${{ matrix.flavor }}
+          path: temp/jest/current/merge/
+      - name: Calculate coverage change
+        id: coverage
+        run: |
+          base_lines_covered=$(jq -r ".total.lines.covered" temp/jest/base/coverage-report.json)
```
Comment (JamesBurnside) [inline .github/workflows/ci.yml:712]: this is a lot in a yaml file (which are notoriously hard to test/debug/reuse) -- can we move this to a script under `common/scripts`?

**Review by JamesBurnside — APPROVED** [review]
LGTM! In the PR desc,. can you mention the impact to CI time this adds?

**Comment by vhuseinova-msft** [comment]
> LGTM! In the PR desc,. can you mention the impact to CI time this adds?

Updated the description, thanks!

**Review by emlynmac — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vhuseinova-msft) [reply .github/workflows/ci.yml]: Updated, but the script is added under workflow folder because of dependencies install

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**